app/controllers/file_manager_controller.rb

```diff
@@ -0,0 +1,228 @@
+# Code for file manager adapted from: adrientoub/file-explorer
+require 'archive'
+require 'pathname'
+
+class FileManagerController < ApplicationController
+  BASE_DIRECTORY = Rails.root.join('courses')
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :update_persistent_announcements
+
+  def index
+    path = params[:path].nil? ? "" : params[:path]
+    absolute_path = check_path_exist(path)
+    if (File.directory?(absolute_path) && check_instructor(absolute_path) ) || path == ""
+      populate_directory(absolute_path, "#{path}/")
+      render 'file_manager/index'
+    elsif File.file?(absolute_path) && check_instructor(absolute_path)
+      if File.size(absolute_path) > 1_000_000 || params[:download]
+        send_file absolute_path
+      else
+        @file = File.read(absolute_path)
+        @path = path
+        render :file, formats: :html
+      end
+    end
+  end
+
+  def upload
+    upload_file(params[:path].nil? ? "" : params[:path])
+  end
+
+  def delete
+    absolute_path = check_path_exist(params[:path])
+    return unless check_instructor(absolute_path)
+
+    absolute_path = check_path_exist(params[:path])
+    current_path = Pathname.new(absolute_path)
+    parent = current_path.parent
+    raise "Unable to delete courses in the root directory." if parent == BASE_DIRECTORY
+
+    FileUtils.rm_rf(absolute_path)
+  end
+
+  def rename
+    absolute_path = check_path_exist(params[:relative_path])
+    if check_instructor(absolute_path)
+      current_path = Pathname.new(absolute_path)
```

Comment:
Once again, `absolute_path` is already a pathname

Reply:
Addressed.